### ISSUE TITLE
Code of Conduct

### ISSUE THREAD

**Issue by Voileexperiments** [Codewars/codewars.com#1440]
From a side tangent at #1438

1. So, first things first, the Code of Conduct is almost invisible. It's in the github repo, not in the main site, and even with that it's *hidden inside the wiki page*. Unless someone's using the github repo often (i.e <1% of the users) they're not going to notice its existence, which makes the whole thing very under-utilized.

2. I see the whole thing resembling a lot from StackOverflow's "be nice" policy, which is also well known for being detrimental, being used as a tool to shoot people you don't like for not respecting some arbitrary measure of manner, and all in all encourages *formalism* more than conveying actual information. 
As an example, let's assume we're talking about #1390 again and we want to add an entry about it to the Code of Conduct. By the current tone of the Code of Conduct, something like this would probably ensue, which is bad because it encourages formalism:
- Complaining about a solution is not encouraged because the resulting manner is bad. Lots of people become really salty when they're complaining about a solution.
The good way would be
- Complaining about a solution is not encouraged because usually little to no information is given about why the solution is not preferred. It does not provide enough information for others to make their own informed judgement on this solution.
Similar, good manners are encouraged not because they're good manners, but because it *reduces friction between communication*.
The thing is, the purpose of CW is (or at least should be) to more about educating people to learn more about programming *than* learn more about manners, and flipping it the other way around is hindering this purpose.

3. For a site that is mostly used for learning, there should be much, *much* more about things like:
- how to ask a question in the comments effectively, e.g not just slapping an issue saying "test is wrong"/"test does not work"
- how to handle enquiries in your kata properly (tip: it's almost always "work together")
- how to make your kata a more fun experience *in general*, e.g good test feedback
- how to make *other's* experience more fun in general: coding should be *fun*, and being petty about compatibility is the opposite to that
- ...

I guess the Code of Conduct is yet another thing that was slapped together quickly (well, just like most things on CW), but since it's actually something that some people will point to it as if it's semi-legally binding inevitably at some point, it needs to be *at the right direction*, which currently doesn't seem to be so. It's also too hidden to be working at all.

**Comment by GiacomoSorbi** [comment]
My 0.02€ here: while I might understand that different cultures have *very* different sensibilities, thus it also pretty hard to figure out what it is straightforward and what is just gratuitously obnoxious, I will both avoid complaining as long as a user is given a feed (no matter how blunt) and try to avoid the personal part out of the way.

Despite being sick and tired with the current trend of most anglo-saxons to be paranoiacally concerned about making other feel "uncomfortable" (which in turns often leads to quick censorship or open witch-hunting, a clear sign those guys never directly experienced a real fascist regimen to know what it means to embrace that kind of thought too easily <_<), I can see how some beginners can quickly get demotivated, particularly if treated as if they were doing the stupidest conceivable thing ever.

Bottom line: as it should be in my interest to give good advice (and it is, as I got some upvotes), I would not start policing CW at all with any kind of code of conduct (more risky than useful, as pointed out); if a user acts as a jerk, he gets downvoted and that's it, as long as no real law is broken.

As easy as that.

[The risk of going stackoverlowing .- if you know what I mean - is still there, but it should not be a major concern atm.]

**Comment by Voileexperiments** [comment]
@GiacomoSorbi Except that the *point* of a Code of Conduct is to not devolve into the situation you're describing. An anarchic system of "vote by the public" is problematic as:

- it focuses on quantity instead of quality, which is the opposite of what learning is like. Not to mention most users in CW are newbies, so essentially you're letting newbies dominate the system
- it reinforces popular opinions very badly. If my read at various places is correct, it's attracting people to do things like:
    - laugh at JS with bogus examples (e.g passing function as arguments with no concern of arity, and then blame JS for their mistakes of passing `parseInt` directly into `map`)
    - complain at short, concise code they don't understand as "clever but disgusting"
    - upvoting every issue regardless of their validity without even reading them because they're also stuck on a kata
and yes, they're horrendous, toxic, and misleading to innocent bystanders. In comparison, being blunt is much milder.
- if you define correctness as how much approval one gets from the people, it basically becomes a popularity contest: instead of talking something that's correct people would be optimizing for getting the most *public appeal*. Basically the opposite of what is intended.
- status quo: How many old katas have dubious top voted solutions because those solutions were the first one being upvoted to the top? And how many new solutions are burried because solution sorting is by default ordered by votes? Also, it's well known that the vote scores will induce bias to how people think of the content in question, which is, again, the opposite of what is wanted. Even in an ideal upvote system you want people to vote according to how they think of the content, and not because "lots of people have upvoted it and so it must be good".

Also, please remember that "being a jerk" is a social concern, and CW is not a social network, so *factual* concern should be put above social concerns. That is, as far as "being a jerk" goes, putting emphasis on it gives the wrong idea that it's more severe than factual concerns. Yes, an obnoxious jerk is *very* annoying, but it's not as damaging as when someone's posting *completely wrong* information, and lots of newbies pick up from it. I think you need to get the priorities right :wink:

**Comment by GiacomoSorbi** [comment]
I am afraid that you took only a part of my thoughts: what you say is mostly right and I tend to agree, pardon me if I gave a different impression.

Not sure why we are putting also old katas and votes in the cauldron (I would rather raise the threshold of honour required to vote a kata, though), but I thought we were just strictly discussing the code of conduct here.

Maybe you misunderstood the point of my mention about upvotes (to positive/constructive comments)?

While I do consider CW as a social network (see [definition](https://en.wikipedia.org/wiki/Social_network), it allows you to manage/share content with other uses, interact with other users, etc), even ignoring its status should not be a point to use in order to legitimise "being a jerk": as a teaching/learning tool, it does not necessarily benefit from being harsher than it is due to point out beginners their own flaws.

A code of conduct should **not** be concerned with telling people what is good code, but, at most, it should have mentions on how to be thankful, leave feeds and contribute.

On the rest, including how poisonous is to be overly-concerned about making everybody happy and comfortable all the time, I think I made my mind clear enough.
